ddtrace/contrib/botocore/patch.py

```diff
@@ -0,0 +1,63 @@
+"""
+Trace queries to aws api done via botocore client
+"""
+
+# project
+from ddtrace import Pin
+
+# 3p
+import wrapt
+import botocore.client
+
+from ...ext import http
+
+# Original botocore client class
+_Botocore_client = botocore.client.BaseClient
+
+DEFAULT_SERVICE = "aws"
+SPAN_TYPE = "http"
+
+
+def patch():
+    # Checking for the version compatibility before patching
+    wrapt.wrap_function_wrapper('botocore.client', 'BaseClient._make_api_call', patched_api_call)
+    Pin(service="botocore", app="botocore", app_type="web").onto(botocore.client.BaseClient)
+
+
+def patched_api_call(original_func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return original_func(*args, **kwargs)
+
+    with pin.tracer.trace('botocore.command', service=DEFAULT_SERVICE, span_type=SPAN_TYPE) as span:
+        _endpoint = getattr(instance, "_endpoint", None)
+        endpoint_name = getattr(_endpoint, "_endpoint_prefix", "unknown")
+        _boto_meta = getattr(instance, "meta", None)
+        region_name = getattr(_boto_meta, "region_name", "unknown")
```

Comment:
using `deep_getattr` may save you some lines https://github.com/DataDog/dd-trace-py/blob/020e9d7a77cb685178509f3e4759bdfd2b26eaee/ddtrace/util.py#L23